### PR TITLE
New parameters in get_last_output_files_* methods

### PR DESCRIPTION
**Pull request by pcharmoille**
adapt to route changes proposed in https://github.com/cgwire/zou/pull/182

### CODE REVIEW

**Comment by frankrousseau** [comment]
Thank you for your PR.

Is it possible to have multiple functions like:

* `all_output_files_for_entity_and_task_type(entity, task_type)`
* `all_output_files_for_entity_and_output_type(entity, output_type)`
* `all_output_files_for_entity_and_representation(entity, represenation)`
* `get_output_file_by_name(entity, name)`

The main idea behind `gazu` is to have functions as straightforward as possible and to do some filtering locally. This one looks complex to deal with.

**Comment by frankrousseau** [comment]
Can you change the PR title?

**Comment by pcharmoille** [comment]
I can add multiple methods redirecting to one with all params

**Comment by frankrousseau** [comment]
Yes it would match better the gazu philosphy. By the way to manage your query parameters, can you use the params field instead of building your own string? The API is cleaner and it manage properly the special characters.

**Comment by pcharmoille** [comment]
Since it will add a lot of methods, I suggest to divide the file module in multiple submodules as :
- workfiles
- outputfiles
- previews
- softwares

You're ok with that ?

**Comment by frankrousseau** [comment]
Can you list the methods you want to add?

Doing this will break many scripts. I'm not sure it's a good thing.

**Comment by pcharmoille** [comment]
I may not have understood but I though you would want :

- `all_output_files_for_entity_and_task_type`
- `all_output_files_for_entity_and_output_type`
- `all_output_files_for_entity_and_representation`
- `all_output_files_for_entity_and_name`

- `all_output_files_for_asset_instance_and_temporal_entity`
- `all_output_files_for_asset_instance_and_task_type`
- `all_output_files_for_asset_instance_and_output_type`
- `all_output_files_for_asset_instance_and_name`
- `all_output_files_for_asset_instance_and_representation`

- `get_last_output_files_for_entity_and_output_type`
(and same for all params)

- `get_last_output_files_for_asset_instance_and_temporal_entity`
(and same for all params)

Is that correct ?

**Comment by frankrousseau** [comment]
That's too much functions. Let's keep it as it is. Sorry.

Can you fix the params thing and use `fetch_all` method instead of `get`?

**Comment by pcharmoille** [comment]
sure

**Comment by pcharmoille** [comment]
fixing the test now

File: gazu/client.py [hunk 1]
```diff
@@ -129,13 +129,14 @@ def get_full_url(path):
...
 
-def get(path, json_response=True, params={}):
+def get(path, json_response=True, params=None):
     """
     Run a get request toward given path for configured host.
 
     Returns:
         The request result.
     """
+    params = params or {}
```
Comment (frankrousseau) [inline gazu/client.py]: Why did you change that?

Comment (pcharmoille) [reply gazu/client.py]: Because it is an anti-pattern to initialise dict in function signatures.
https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html

I can leave it but it may lead to "unforeseen consequences"

**Comment by pcharmoille** [comment]
I added a `build_path_with_params` method in the client so we can reproduce the behaviour of url construction in tests instead of writing the full path.

**Comment by frankrousseau** [comment]
Thank you for this addition. I will release it when we will be done with the Zou PR.
